vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -634,6 +634,80 @@ testLtsKernel() {
 
 }
 
+testAutologinDisabled() {
+  local test="testAutologinDisabled"
+  os_sku=$1
+  echo "$test:Start"
+
+  if [ "$os_sku" = "Flatcar" ]; then
+    local failed=0
+
+    # Test 1: Check actual behavior using loginctl
+    # With autologin: we should see sessions with Remote=no and Service=login
+    # Without autologin: we should see only SSH sessions (Remote=yes) or no sessions
+    echo "$test: Checking for console autologin sessions using loginctl"
+
+    local autologin_session_ids=""
+    autologin_session_ids=$(loginctl list-sessions --no-legend 2>/dev/null | while read -r session_id rest; do
+      if [ -n "$session_id" ]; then
+        local session_info
+        session_info=$(loginctl show-session "$session_id" 2>/dev/null || true)
+        local is_remote=$(echo "$session_info" | sed -n "s/^Remote=//p")
+        local service=$(echo "$session_info" | sed -n "s/^Service=//p")
+
+        # Console autologin sessions have Remote=no and Service=login
+        if [ "$is_remote" = "no" ] && [ "$service" = "login" ]; then
+          echo "$session_id"
+        fi
+      fi
+    done)
+
+    if [ -n "$autologin_session_ids" ]; then
+      err $test "Found console autologin session(s) with Remote=no and Service=login"
+      echo "$test: Autologin session IDs: $autologin_session_ids" >&2
+      echo "$test: All sessions:" >&2
+      loginctl list-sessions 2>/dev/null >&2 || true
+      failed=1
+    else
+      echo "$test: No console autologin sessions found"
+    fi
+
+    # Test 2: Check for running agetty processes
+    # When autologin is disabled, agetty processes wait for login
+    # When autologin is enabled, getty completes immediately and hands off to login
+    echo "$test: Checking for agetty processes"
+    if pgrep -x agetty >/dev/null 2>&1; then
+      echo "$test: Found agetty processes waiting for login (autologin disabled, correct)"
+    else
+      err $test "No agetty processes found - getty may have completed due to autologin"
+      failed=1
+    fi
+
+    # Test 3: Check kernel parameter configuration
+    echo "$test: Checking kernel command line for flatcar.autologin parameter"
+    if grep -q "flatcar.autologin" /proc/cmdline; then
+      err $test "flatcar.autologin kernel parameter found in /proc/cmdline but should be absent for AKS security compliance"
+      echo "$test: Full kernel command line:" >&2
+      cat /proc/cmdline >&2
+      failed=1
+    else
+      echo "$test: flatcar.autologin parameter correctly absent from kernel command line"
+    fi
+
+    if [ $failed -eq 1 ]; then
```

Comment:
Probably easier to abort when the error occurs rather than having a "failed" variable. Then you can include a little more information in the error message